rename_imports.py

```diff
@@ -0,0 +1,78 @@
+
+from LSP.plugin.core.file_watcher import FileWatcherEvent, FileWatcherProtocol, get_file_watcher_implementation, FileWatcher
+from LSP.plugin.core.typing import  List
+import os
+import sublime
+
+
+def register_rename_import_file_watcher(root_path: str):
+    file_watcher = get_file_watcher_implementation()
+    if file_watcher:
+        watcher = file_watcher.create(
+            root_path=root_path,
+            patterns=["**/*.ts", "**/*.tsx", "**/*.js", "**/*.jsx"],
+            events=["create", "delete"],
+            ignores=['**/node_modules/**'],
+            handler=HandleRenameImport
+        )
+
+
+queued_events = []  # type: List[FileWatcherEvent]
+
+
+class HandleRenameImport(FileWatcherProtocol):
+    def on_file_event_async(events: List[FileWatcherEvent]) -> None:
+        global queued_events
+        queued_events.append(events[0])
+        sublime.set_timeout(check_if_there_was_a_rename, 10)
+
+
+def check_if_there_was_a_rename():
+    global queued_events
+    if len(queued_events) == 2:
+        first_event, new_name = queued_events[0]
+        second_event, old_name = queued_events[1]
+        if (first_event == 'create' and second_event == 'delete'):
```

Comment:
This order is not guaranteed. I believe I got an opposite on Mac when testing this.

Reply:
Fixed with explicitly finding those events:
```
create_event = next((event for event in self.events if event[0] == 'create'), None)
delete_event = next((event for event in self.events if event[0] == 'delete'), None)
if (create_event and delete_event):
```